hal/cpu_uid_api.h

```diff
@@ -0,0 +1,61 @@
+/** \addtogroup hal */
+/** @{*/
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_CPU_UID_API_H
+#define MBED_CPU_UID_API_H
+
+#include <stddef.h>
+#include "device.h"
+
+#if DEVICE_CPUUID
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * \defgroup hal_cpu_uid CPUUID hal functions
```

Comment:
@bremoran @seankinghan @LiyouZhou Do we want to have some minimal require lenght?

Reply:
For the CPU UID? I don't know whether that would be useful or not. The UID isn't really important from a security perspective. If the device has an encryption root, that's a whole other story.